apis/r/R/PlatformConfig.R

```diff
@@ -0,0 +1,130 @@
+      return(super$get(key = platform)$keys())
+    },
+    #' @param platform The name of the \dQuote{platform} (outer key) to fetch
+    #' @param op The name of the \dQuote{option} of \code{platform} to fetch;
+    #' if \code{NULL}, returns the \link[tiledbsoma:ConfigList]{configuration}
+    #' for \code{platform}
+    #' @param key The \dQuote{key} (inner key) for \code{op} in
+    #' \code{platform} to fetch; if \code{NULL} and \code{op} is passed,
+    #' returns the \link[tiledbsoma:ScalarMap]{map} for \code{op}
+    #' in \code{platform}
+    #' @templateVar key key
+    #' @templateVar default null
+    #' @template param-default
+    #'
+    #' @return The value of \code{key} for \code{op} in \code{platform} in the
+    #' map, or \code{default} if \code{key} is not found
+    #'
+    get = function(
+      platform,
+      op = NULL,
+      key = NULL,
+      default = rlang::missing_arg()
+    ) {
+      if (!length(x = self)) {
+        warning("No platforms configured", call. = FALSE)
+        return(NULL)
+      }
+      platform <- platform[1L] %||% self$platforms()[1L]
+      platform <- tryCatch(
+        expr = match.arg(arg = platform, choices = self$platforms()),
+        error = \(...) NULL
+      )
+      if (is.null(x = platform)) {
+        if (rlang::is_missing(x = default)) {
+          private$.key_error(key = platform)
+        }
+        return(default)
+      }
+      pmap <- super$get(key = platform)
+      if (is.null(x = op)) {
+        return(pmap)
+      }
+      return(pmap$get(op = op, key = key, default = default))
+      # opmap <- pmap$get(key = op)
+      # if (is.null(x = key)) {
+      #   return(opmap)
+      # }
+      # return(opmap$get(key = key, default = default))
+    },
+    #' @param platform The name of the \dQuote{platform} (outer key) to fetch
+    #'
+    #' @return The \code{\link{ConfigList}} for \code{platform}
+    #'
+    get_ops = function(platform) {
+      platform <- platform[1L] %||% self$platforms()[1L]
+      platform <- match.arg(arg = platform, choices = self$platforms())
+      return(super$get(key = platform))
+    },
+    #' @param platform The name of the \dQuote{platform} (outer key) to set
+    #' @param op Name of the \dQuote{option} (middile key) in \code{platform}
+    #' to set
+    #' @param key Inner key to set
+    #' @param value Value to add for \code{key}, or \code{NULL} to remove the
+    #' entry for \code{key}; optionally provide only \code{platfomr}, \code{op},
+    #' and \code{value} as a \code{\link{ScalarMap}} to update \code{op} for
+    #' \code{platform} with the keys and values from \code{value}
+    #'
+    #' @return \[chainable\] Invisibly returns \code{self} with \code{value}
+    #' added for \code{key} in \code{op} for \code{platform}
+    #'
+    set = function(platform, op, key, value) {
+      stopifnot(
+        is.character(x = platform) && length(x = platform) == 1L,
+        is.character(x = op) && length(x = op) == 1L
```

Comment:
You could use the form  `stopifnot("text to be displayed if error" = test1 && test2)` which is nicer on the user.  Even if it is stricly an internal `assert()` style test it still provides comments.